newIDE/app/src/EventsSheet/ParameterFields/KeyboardKeyField.js

```diff
@@ -162,18 +173,15 @@ export default React.forwardRef<ParameterFieldProps, ParameterFieldInterface>(
 
     // If the current value is not in the list, display an expression field.
     const [isExpressionField, setIsExpressionField] = React.useState(
-      !!props.value && !isKeyValid(props.value)
+      !!props.value && !isValidLiteralKeyboardKey(props.value)
     );
 
     const switchFieldType = () => {
       setIsExpressionField(!isExpressionField);
     };
 
     const onChangeSelectValue = (value: string) => {
-      const oldValue = getStringContent(props.value);
-      if (oldValue !== value) {
-        props.onChange(valueRegex.test(value) ? `"${value}"` : value);
-      }
+      props.onChange(valueRegex.test(value) ? `"${value}"` : value);
```

Comment:
Checking that the value changed was necessary and it was forbidding to go from `Down` to `"Down"`. It's a case that will happens when migrating from the deprecated conditions.